netmiko/fortinet/fortinet_ssh_ibmcloud.py

```diff
@@ -0,0 +1,220 @@
+            self._test_channel_read(pattern=self.prompt_pattern)
+
+        self.set_base_prompt()
+        self._vdoms = self._vdoms_enabled()
+        self._os_version = self._determine_os_version()
+        # Retain how the 'output mode' was original configured.
+        self._original_output_mode = self._get_output_mode()
+        self._output_mode = self._original_output_mode
+        self.disable_paging()
+
+    def set_base_prompt(
+        self,
+        pri_prompt_terminator: str = r"#",
+        alt_prompt_terminator: str = r"$",
+        delay_factor: float = 1.0,
+        pattern: Optional[str] = None,
+    ) -> str:
+        if not pattern:
+            pattern = self.prompt_pattern
+        return super().set_base_prompt(
+            pri_prompt_terminator=pri_prompt_terminator,
+            alt_prompt_terminator=alt_prompt_terminator,
+            delay_factor=delay_factor,
+            pattern=pattern,
+        )
+
+    def find_prompt(
+        self, delay_factor: float = 1.0, pattern: Optional[str] = None
+    ) -> str:
+        if not pattern:
+            pattern = self.prompt_pattern
+        return super().find_prompt(
+            delay_factor=delay_factor,
+            pattern=pattern,
+        )
+
+    def _vdoms_enabled(self) -> bool:
+        """Determine whether virtual domains are enabled or not."""
+        check_command = "get system status | grep Virtual"
+        output = self._send_command_str(
+            check_command, expect_string=self.prompt_pattern
+        )
+        return bool(
+            re.search(r"Virtual domain configuration: (multiple|enable)", output)
+        )
+
+    def _config_global(self) -> str:
+        """Enter 'config global' mode, raise a ValueError exception on failure."""
+        try:
+            return self._send_command_str(
+                "config global", expect_string=self.prompt_pattern
+            )
+        except Exception:
+            msg = """
+Netmiko may require 'config global' access to properly disable output paging.
+Alternatively you can try configuring 'configure system console -> set output standard'.
+"""
+            raise ValueError(msg)
+
+    def _exit_config_global(self) -> str:
+        """Exit 'config global' mode."""
+        try:
+            return self._send_command_str("end", expect_string=self.prompt_pattern)
+        except Exception:
+            msg = "Unable to properly exit 'config global' mode."
+            raise ValueError(msg)
+
+    def disable_paging(
+        self,
+        command: str = "terminal length 0",
+        delay_factor: Optional[float] = None,
+        cmd_verify: bool = True,
+        pattern: Optional[str] = None,
+    ) -> str:
+        """Disable paging is only available with specific roles so it may fail."""
+
+        output = ""
+        if self._output_mode == "standard":
+            # Do nothing - already correct.
+            return ""
+
+        if self._vdoms:
+            output += self._config_global()
```

Comment:
I didn't understand this? I thought your purpose was to get rid of `config global`, but it still does `config global` (depending on VDoms)?